sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/BasicWriteStatsTracker.scala

```diff
@@ -166,7 +166,7 @@ class BasicWriteTaskStatsTracker(hadoopConf: Configuration)
     }
 
     if (numSubmittedFiles != numFiles) {
-      logInfo(s"Expected $numSubmittedFiles files, but only saw $numFiles. " +
+      logWarning(s"Expected $numSubmittedFiles files, but only saw $numFiles. " +
```

Comment:
WDYT @steveloughran ?

Reply:
This used to happen sometimes on code writing files to S3 where a cached HEAD-generated 404 was being returned even after the upload.
* Not going to be an issue on S3 any more, never was on GCS/Azure. 
* s3a magic committer creates marker files for this code
* Are there any FileOutputFormat's which don't write zero byte files if the client didn't write the output?